src/vaadin-message-list.js

```diff
@@ -118,6 +131,76 @@ class MessageListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     }
   }
 
+  /**
+   * @param {!KeyboardEvent} event
+   * @protected
+   */
+  _onKeydown(event) {
+    if (event.metaKey || event.ctrlKey) {
+      return;
+    }
+
+    // get index of the item that was focused when event happened
+    const target = event.composedPath()[0];
+    let currentIndex = this._messages.indexOf(target);
+
+    switch (event.key) {
+      case 'ArrowUp':
+        currentIndex--;
+        break;
+      case 'ArrowDown':
+        currentIndex++;
+        break;
+      case 'Home':
+        currentIndex = 0;
+        break;
+      case 'End':
+        currentIndex = this._messages.length - 1;
+        break;
+      default:
+        return; // nothing to do
+    }
+    if (currentIndex < 0) {
+      currentIndex = this._messages.length - 1;
+    }
+    if (currentIndex > this._messages.length - 1) {
+      currentIndex = 0;
+    }
+    this._focus(currentIndex);
+    event.preventDefault();
+  }
+
+  /**
+   * @param {number} idx
+   * @protected
+   */
+  _focus(idx) {
+    const target = this._messages[idx];
+    target.focus();
+  }
+
+  _handleFocusEvent(e) {
+    this._setTabIndexesByMessage(
+      e.composedPath().filter(elem => elem.tagName === 'VAADIN-MESSAGE')[0]);
```

Comment:
```suggestion
e.composedPath().filter((elem) => elem.tagName.toLowerCase() === 'vaadin-message')[0]);
```

Reply:
I changed this to  `elem.tagName && elem.tagName.toLowerCase()` as there seems to be elements without a tagName.